plugins/pagerduty-incidents/src/components/PagerDutyIncidents.tsx

```diff
@@ -0,0 +1,384 @@
+import React, { useState, useEffect, useMemo } from "react";
+import { PluginContextLocation } from "@cortexapps/plugin-core";
+import {
+  SimpleTable,
+  // Text,
+  Title,
+  Loader,
+  usePluginContext,
+} from "@cortexapps/plugin-core/components";
+
+import {
+  Avatar,
+  AvatarGroup,
+  Box,
+  Badge,
+  Flex,
+  // Center,
+  Text,
+} from "@chakra-ui/react";
+
+import "../baseStyles.css";
+
+interface PagerDutyIncidentsProps {
+  entityYaml: Record<string, any>;
+}
+
+const StatusBadge: React.FC<{ status: string }> = ({ status }) => {
+  const color =
+    status === "resolved"
+      ? "green"
+      : status === "acknowledged"
+      ? "yellow"
+      : "red";
+  return (
+    <Badge p={1} colorScheme={color} variant="subtle">
+      {status}
+    </Badge>
+  );
+};
+
+const UrgencyBadge: React.FC<{ urgency: string }> = ({ urgency }) => {
+  const variant = urgency === "high" ? "solid" : "subtle";
+  return (
+    <Badge p={1} variant={variant}>
+      {urgency}
+    </Badge>
+  );
+};
+
+const PagerDutyIncidents: React.FC<PagerDutyIncidentsProps> = ({
+  entityYaml,
+}) => {
+  const [hasIncidents, setHasIncidents] = useState(false);
+  const context = usePluginContext();
+
+  const [service, setService] = useState(null as any);
```

Comment:
See below comment for 76, if we go that route these will change to pass isLoading to each component reference as well as error state, and replace the state setter with your custom hook